3scaleAdapter/pkg/threescale/metrics/prometheus.go

```diff
@@ -79,6 +89,19 @@ func (r *Reporter) ObserveBackendLatency(backendURL string, serviceID string, ob
 	}
 }
 
+// ReportBackendStatus reports a hit to 3scale backend and reports status code of the result
+func (r *Reporter) ReportBackendStatus(code int, backendURL string, serviceID string) error {
+	codeStr := strconv.Itoa(code)
+	if len(codeStr) != 3 {
+		return errors.New("invalid status code reported")
+	}
+
+	if r != nil && r.shouldReport {
```

Comment:
is it possible for `r` to be `nil` and have this method called?

Reply:
Yep I believe so. https://play.golang.org/p/3ki9xyzhH3